hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -426,4 +434,41 @@ public boolean isValidNode(DatanodeDetails datanodeDetails,
     }
     return false;
   }
+
+  @Override
+  public Set<ContainerReplica> replicasToCopy(
+         Set<ContainerReplica> replicas, int expectedCountPerUniqueReplicas,
+         int expectedUniqueGroups) {
+    Map<Node, Set<ContainerReplica>> placementGroupReplicaIdMap
```

Comment:
I think it would make the for loop below easier if we transformed this into a `Map<Node, List<ContainerReplica>>`. It would probably be slightly faster too, as forming a set is more expensive that a list.

As we are passed a Set of ContainerReplica, we already know there are no duplicates, so its safe enough to transform to the List.

Reply:
Done